src/core/config/local_config.ts

```diff
@@ -0,0 +1,188 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { IsEmail, IsNotEmpty, IsObject, IsString, validateSync } from 'class-validator'
+import { type ListrTask } from 'listr2'
+import fs from 'fs'
+import * as yaml from 'yaml'
+import { flags } from '../../commands/index.js'
+import { type ClusterMapping, type Deployment, type Deployments, type LocalConfigData } from './local_config_data.js'
+import { MissingArgumentError, SoloError } from '../errors.js'
+import { promptDeploymentClusters, promptDeploymentName, promptUserEmailAddress } from '../../commands/prompts.js'
+import { type SoloLogger } from '../logging.js'
+import { Task } from '../task.js'
+
+export class LocalConfig implements LocalConfigData {
+    @IsNotEmpty()
+    @IsEmail()
+    userEmailAddress: string
+
+    // The string is the name of the deployment, will be used as the namespace,
+    // so it needs to be available in all targeted clusters
+    @IsNotEmpty()
+    @IsObject()
+    deployments: Deployments
+
+    @IsNotEmpty()
+    @IsString()
+    currentDeploymentName : string
+
+    // contextName refers to the "CURRENT NAME", and clusterName refers to the CLUSTER leveraged in kubeConfig.currentContext
+    // { clusterName : string, contextName : string }
+    @IsNotEmpty()
+    @IsObject()
+    clusterMappings: ClusterMapping
+
+    private readonly skipPromptTask: boolean = false
+
+    constructor (private readonly filePath: string, private readonly logger: SoloLogger) {
+        if (!filePath || filePath === '') throw new MissingArgumentError('a valid filePath is required')
+        if (!logger) throw new Error('An instance of core/SoloLogger is required')
+
+        const allowedKeys = ['userEmailAddress', 'deployments', 'currentDeploymentName', 'clusterMappings']
+        if (this.configFileExists()) {
+            const fileContent = fs.readFileSync(filePath, 'utf8')
+            const parsedConfig = yaml.parse(fileContent)
+
+            for(const key in parsedConfig) {
+                if (!allowedKeys.includes(key)) {
+                    throw new SoloError('Validation of local config failed')
+                }
+                this[key] = parsedConfig[key]
+            }
+
+            this.validate()
+            this.skipPromptTask = true
+        }
+    }
+
+    private validate () {
+        const genericMessage = 'Validation of local config failed'
+        const errors = validateSync(this, {})
+
+        if (errors.length) {
+            throw new SoloError(genericMessage)
+        }
+
+        try {
+            // Custom validations:
+            for (const deploymentName in this.deployments) {
+                const deployment = this.deployments[deploymentName]
+                const deploymentIsObject = deployment && typeof deployment === 'object'
```

Comment:
shouldn't it be of type Deployment?  is the class-transformer working correctly?

Reply:
this block is deleted now